src/overlays/actors/ovl_En_Syateki_Niw/z_en_syateki_niw.c

```diff
@@ -21,37 +43,707 @@ const ActorInit En_Syateki_Niw_InitVars = {
+                this->unk_25E = this->unk_260;
+                break;
+        }
+    }
+}
+
+#ifdef NON_MATCHING
+// The switch
+void EnSyatekiNiw_Update(Actor* thisx, GlobalContext* globalCtx) {
+    s16 i;
+    s32 tmp;
+    EnSyatekiNiw* this = THIS;
+    // D_80B136B8
+    Vec3f sp90 = { 0.0f, 0.0f, 0.0f };
+    // D_80B136C4
+    Vec3f sp84 = { 0.0f, 0.0f, 0.0f };
+    Vec3f sp78;
+    Vec3f sp6C;
+    Vec3f sp60;
+
+    func_80B132A8(this, globalCtx);
+    this->unk_28C++;
+    if (this->unk_254 != 0) {
+        this->unk_254--;
+    }
+    if (this->unk_258 != 0) {
+        this->unk_258--;
+    }
+    if (this->unk_25A != 0) {
+        this->unk_25A--;
+    }
+    if (this->unk_25C != 0) {
+        this->unk_25C--;
+    }
+    if (this->unk_25E != 0) {
+        this->unk_25E--;
+    }
+    if (this->unk_262 != 0) {
+        this->unk_262--;
+    }
+    if (this->unk_260 != 0) {
+        this->unk_260--;
+    }
+    this->actor.shape.rot = this->actor.posRot.rot;
+    this->actor.shape.unk_10 = 15.0f;
+
+    this->actionFunc(this, globalCtx);
+    Actor_MoveForward(&this->actor);
+    func_8002E4B4(globalCtx, &this->actor, 20.0f, 20.0f, 60.0f, 0x1D);
+
+    if (this->unk_2A0 != 0) {
+        for (i = 0; i < 20; i++) {
+            sp78.x = Math_Rand_CenteredFloat(10.0f) + this->actor.posRot.pos.x;
+            sp78.y = Math_Rand_CenteredFloat(10.0f) + (this->actor.posRot.pos.y + 20.0f);
+            sp78.z = Math_Rand_CenteredFloat(10.0f) + this->actor.posRot.pos.z;
+            sp6C.x = Math_Rand_CenteredFloat(3.0f);
+            sp6C.y = (Math_Rand_ZeroFloat(2.0f) * 0.5f) + 2.0f;
+            sp6C.z = Math_Rand_CenteredFloat(3.0f);
+            sp60.z = sp60.x = 0.0f;
+            sp60.y = -0.15f;
+            func_80B131B8(this, &sp78, &sp6C, &sp60, Math_Rand_ZeroFloat(8.0f) + 8.0f);
+        }
+        this->unk_2A0 = 0;
+    }
+
+    func_80B12BA4(this, globalCtx);
+    if (this->unk_262 == 0) {
+        if (this->actionFunc == func_80B11E78) {
+            this->unk_262 = 0x12C;
+            Audio_PlayActorSound2(&this->actor, NA_SE_EV_CHICKEN_CRY_N);
+        } else {
+            this->unk_262 = 0x1E;
+            Audio_PlayActorSound2(&this->actor, NA_SE_EV_CHICKEN_CRY_A);
+        }
+    }
+
+    tmp = 0;
+    switch (this->unk_29E) {
+        case 0:
+            if (globalCtx->unk_11E5C != 0) {
+                tmp = 1;
+            }
+            break;
+        case 1:
+            tmp = 1;
+            break;
+    }
+
+    if (tmp != 0) {
+        Collider_CylinderUpdate(&this->actor, &this->collider);
+        CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+        CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    }
+}
+#else
 #pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Syateki_Niw/EnSyatekiNiw_Update.s")
+#endif
+
+s32 func_80B12FE0(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* thisx) {
```

Comment:
Probably should be `EnSyatekiNiw_OverrideLimbDraw`

Reply:
yeah, rename this